packages/skott/src/skott.ts

```diff
@@ -485,12 +528,133 @@ export class Skott<T> {
     };
   }
 
+  private getValidGroup(nodePath: string) {
+    if (!this.config.groupBy) {
+      throw new Error(
+        "Grouped graph can't be built without a groupBy function in the configuration"
+      );
+    }
+
+    const result = this.config.groupBy(nodePath);
+
+    if (typeof result === "string" || !result) {
+      return result;
+    }
+
+    throw new Error(
+      `groupBy function must return a string or undefined, but returned "${result}" (for "${nodePath}")`
+    );
+  }
+
+  private addNodeToGroupedGraph(node: SkottNode<T>) {
+    const group = this.getValidGroup(node.id);
+
+    if (group) {
+      /**
+       * Typecast is ok here:
+       * current typings are expecting SkottNodeBody + other arbitary values if available, but currently TS is not happy with it.
+       *
+       * The SkottNode type should be described in some other way to properly support that, but SkottNodeBody is avaiable at the runtime anyway.
+       */
+      const nodeBody = node.body as SkottNodeBody;
```

Comment:
It seems that `nody.body` without the type casting works for me, TypeScript is not complaining as the return type looks like the same as `SkottNodeBody`

Reply:
Not sure what changed, but TypeScript is indeed does not complain to me about `node.body` type anymore 🤔 

Removed the typecast